### PR TITLE
Upgrade spider-gazelle

### PR DESCRIPTION
**Pull request by stakach**
Uses the latest version of crystal lang and spider-gazelle

### CODE REVIEW

File: crystal/spider-gazelle/Dockerfile [hunk 1]
```diff
@@ -8,4 +8,4 @@ COPY src src
 
 RUN shards build --release --no-debug
 
-CMD bin/app
+CMD bin/app -w $(nproc)
```
Comment (waghanza) [inline crystal/spider-gazelle/Dockerfile]: :+1:

File: crystal/spider-gazelle/shard.yml [hunk 2]
```diff
@@ -4,10 +4,10 @@ version: 1.0.0
 dependencies:
   action-controller:
     github: spider-gazelle/action-controller
-    version: ~> 1.0.5
+    version: "~> 1.1"
```
Comment (waghanza) [inline crystal/spider-gazelle/shard.yml]: @stakach I prefer to use X.Y.Z to defines deps

**Comment by waghanza** [comment]
On my `workstation` I have

| |  50th percentile |  90th percentile |  99th percentile | 99.9th percentile | Standard deviation | Requests / s | Throughput |
|---|---|---|---|---|---|---|---|
| master | 20.86 ms | 26.61 ms | 28.83 ms | 217.38 ms | 5119.33 | 46156.00 | 49.74 MB |
| here | 6.16 ms | 11.39 ms | 19.04 ms | 35.05 ms | 3470.00 | 139839.00 | 149.54 MB |

well done @stakach :tada:

**Comment by stakach** [comment]
Woot!

**Comment by waghanza** [comment]
@stakach what changes where made for this release of `spider-gazelle`. Is there a framework changelog I can fond ?

**Comment by stakach** [comment]
No major changes outside of crystal lang compatibility and the addition of clustering / worker support.
Which basically amounts to the addition of this function: https://github.com/spider-gazelle/action-controller/blob/master/src/action-controller/server.cr#L71

So simple to some powerful things in Crystal

**Comment by rishavs** [comment]
@stakach 
considering that a lot of people are still waiting for parallelism in Crystal, can you write a short article on your clustering implementation? It would help the community a lot.
I tried to go through the code but much of it is beyond me.

**Comment by stakach** [comment]
Sure, I’ll add a doc to the repo, in the mean time what it is doing is:
• saving the command line arguments
• launching new child processes without the cluster arg (new process vs forking for windows compatibility)
• child process PIDs are saved
• child processes inherit stdout and stderr
• the socket binding is reused between the processes
• ctrl-c Signal is sent to parent and children processes
• children shutdown gracefully 
• parent waits for all the children to shutdown before also shutting down gracefully

**Comment by stakach** [comment]
The write up: https://github.com/spider-gazelle/action-controller/blob/master/clustering.md
Hope that helps!

**Pull request merged**